src/Propulsion/Streams.fs

```diff
@@ -284,9 +284,9 @@ module Scheduling =
         let markBusy stream = busy.Add stream |> ignore
         let markNotBusy stream = busy.Remove stream |> ignore
 
-        member _.ChooseDispatchable(s: FsCodec.StreamName, allowGaps): StreamState<'Format> voption =
+        member _.ChooseDispatchable(s: FsCodec.StreamName, requireAll): StreamState<'Format> voption =
             match tryGetItem s with
-            | ValueSome ss when ss.IsReady && (allowGaps || not ss.HasGap) && not (busy.Contains s) -> ValueSome ss
+            | ValueSome ss when ss.HasValidEvents && (not requireAll || ss.IsComplete) && not (busy.Contains s) -> ValueSome ss
```

Comment:
Not sure about removing the gap term 🤔

Reply:
Yeah, but Gap is used in the cosmos source as the gap from checkpoint to tail
and the the issue here is that there's both Gaps and starting at a position other than 0 in play as a consideration
If we want a generic one word term, maybe that's waiting or something

Reply:
added a commit to rename to waiting as that was already used in the log messages and is not a worse term than gapped